code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -18,6 +18,17 @@
 	..()
 	handle_layer()
 
+//For unanchored/moving chairs: We aren't dense, so let's be a little pickier on what we can and can't pass through. Mobs can't walk through us, so let's not walk through them.
+//NOTE: Vehicles are "moving chairs", but they shouldn't be able to move through people either! The firebird can do this, because pass_flags override cantpass_flags.
+/obj/structure/bed/chair/lock_atom(var/atom/movable/AM)
```

Comment:
May implement this with a new flag for locking datums, similar to DENSE_WHEN_LOCKING

Reply:
Yeah this should definitely use locking categories, if some ghosts locks himself to a chair or something it's gonna break.

Reply:
Agreed, it's bad, will use something different